core/src/main/scala/cell/HandlerPool.scala

```diff
@@ -205,6 +250,31 @@ class HandlerPool(parallelism: Int = 8, unhandledExceptionHandler: Throwable =>
     })
   }
 
+  /**
+   * If a cell is triggered, it's `init` method is
+   * run to both get an initial (or possibly final) value
+   * and to set up dependencies. All dependees automatically
+   * get triggered.
+   *
+   * @param cell The cell that is triggered.
+   */
+  def triggerExecution[K <: Key[V], V](cell: Cell[K, V]): Unit = {
+    if (cell.markAsRunning())
+      execute(() => {
+        val completer = cell.asInstanceOf[CellImpl[K, V]]
+        val outcome = completer.init()
+        outcome match {
+          case Outcome(v, isFinal) =>
+            completer.put(v, isFinal)
```

Comment:
This can throw an IllegalStateException, if the cell has already been completed in `init`. (For example, it could add a dependency that gets triggered concurrently and returns a FinalOutcome.) Maybe we want to catch this exception here and not pass it to the `unhandledExceptionHandler`?

Reply:
Catch this exception, because a different order of execution would not raise this exception (see the added test "init: concurrent completion").